sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/main/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubsMessageChannelBinder.java

```diff
@@ -117,8 +115,6 @@ protected MessageHandler createProducerMessageHandler(
         if (producerProperties.isPartitioned()) {
             handler.setPartitionIdExpression(
                 EXPRESSION_PARSER.parseExpression("headers['" + BinderHeaders.PARTITION_HEADER + "']"));
-        } else {
-            handler.setPartitionKeyExpression(new FunctionExpression<Message<?>>(m -> m.getPayload().hashCode()));
         }
```

Comment:
Please check whether this change is necessary for the [Service Bus binder](https://github.com/Azure/azure-sdk-for-java/blob/cbbf4ba4b730c379c8b1091f2f1e3841ca323708/sdk/spring/spring-cloud-azure-stream-binder-servicebus/src/main/java/com/azure/spring/cloud/stream/binder/servicebus/ServiceBusMessageChannelBinder.java#L122).